includes/http/WP_AI_Client_Client_Adapter.php

```diff
@@ -0,0 +1,192 @@
+<?php
+/**
+ * WordPress AI Client HTTP Client Adapter
+ *
+ * @package WordPress\AI_Client
+ * @since n.e.x.t
+ */
+
+namespace WordPress\AI_Client\HTTP;
+
+use Psr\Http\Client\ClientInterface;
+use Psr\Http\Message\RequestInterface;
+use Psr\Http\Message\ResponseInterface;
+use Psr\Http\Message\ResponseFactoryInterface;
+use Psr\Http\Message\StreamFactoryInterface;
+
+/**
+ * PSR-18 HTTP Client adapter using WordPress HTTP API
+ *
+ * This adapter allows WordPress HTTP functions to be used
+ * as a PSR-18 compliant HTTP client.
+ *
+ * @since n.e.x.t
+ */
+class WP_AI_Client_Client_Adapter implements ClientInterface {
+
+	/**
+	 * Response factory instance.
+	 *
+	 * @var ResponseFactoryInterface
+	 */
+	private $response_factory;
+
+	/**
+	 * Stream factory instance.
+	 *
+	 * @var StreamFactoryInterface
+	 */
+	private $stream_factory;
+
+	/**
+	 * Constructor.
+	 *
+	 * @param ResponseFactoryInterface $response_factory PSR-17 Response factory.
+	 * @param StreamFactoryInterface   $stream_factory   PSR-17 Stream factory.
+	 */
+	public function __construct( ResponseFactoryInterface $response_factory, StreamFactoryInterface $stream_factory ) {
+		$this->response_factory = $response_factory;
+		$this->stream_factory   = $stream_factory;
+	}
+
+	/**
+	 * Sends a PSR-7 request and returns a PSR-7 response.
+	 *
+	 * @param RequestInterface $request The PSR-7 request.
+	 *
+	 * @return ResponseInterface The PSR-7 response.
+	 *
+	 * @throws \Exception If the WordPress HTTP request fails.
+	 */
+	public function sendRequest( RequestInterface $request ): ResponseInterface {
+		$args = $this->prepare_wp_args( $request );
+		$url  = (string) $request->getUri();
+
+		/** Ignoring PHPStan for WordPress-specific array structure. @phpstan-ignore-next-line */
+		$response = \wp_remote_request( $url, $args );
+
+		if ( \is_wp_error( $response ) ) {
+			// TODO: Update to use PHP AI Client exceptions.
+			throw new \Exception(
+				$response->get_error_message(), // phpcs:ignore WordPress.Security.EscapeOutput.ExceptionNotEscaped
+				$response->get_error_code() ? (int) $response->get_error_code() : 0
+			);
+		}
+
+		return $this->create_psr_response( $response );
+	}
+
+	/**
+	 * Prepare WordPress HTTP API arguments from PSR-7 request.
+	 *
+	 * @param RequestInterface $request The PSR-7 request.
+	 *
+	 * @return array<string, mixed> WordPress HTTP API arguments.
+	 */
+	private function prepare_wp_args( RequestInterface $request ): array {
+		$args = array(
+			'method'      => $request->getMethod(),
+			'headers'     => $this->prepare_headers( $request ),
+			'body'        => $this->prepare_body( $request ),
+			'timeout'     => 30,
+			'redirection' => 5,
```

Comment:
This is really not great - it's not a problem of this PR, but something we need to fix soon in https://github.com/WordPress/php-ai-client: It needs to be possible for a request to specify additional options for how the request should be sent.

I believe we talked about this a few weeks back but decided to defer it. Now that 0.1.0 is out though, it's an important limitation to address. There are many AI calls for which whatever the default timeout of different client implementations is is not sufficient.

Reply:
Just opened https://github.com/WordPress/php-ai-client/pull/109 to add the feature!